src/modules/previewpane/SvgPreviewHandler/SvgPreviewControl.cs

```diff
@@ -26,30 +45,82 @@ public override void DoPreview<T>(T dataSource)
+                using (var stream = new StreamWrapper(dataSource as IStream))
+                {
+                    using (var reader = new StreamReader(stream))
+                    {
+                        svgData = reader.ReadToEnd();
+                    }
+                }
+
+                svgData = string.IsNullOrWhiteSpace(svgData) ? svgData : SvgPreviewHandlerHelper.RemoveElements(svgData, out this.foundFilteredElements);
+                if (this.foundFilteredElements)
+                {
+                    this.AddTextBoxControl();
+                }
+
+                this.AddBrowserControl(svgData);
+                this.Resize += this.FormResized;
                 base.DoPreview(dataSource);
             });
         }
 
         /// <summary>
-        /// Free resources on the unload of Preview.
+        /// Occurs when RichtextBox is resized.
         /// </summary>
-        public override void Unload()
+        /// <param name="sender">Reference to resized control.</param>
+        /// <param name="e">Provides data for the ContentsResized event.</param>
+        private void RTBContentsResized(object sender, ContentsResizedEventArgs e)
         {
-            base.Unload();
-            if (this.dataSourceStream != null)
+            var richTextBox = sender as RichTextBox;
+            richTextBox.Height = e.NewRectangle.Height + 5;
+        }
+
+        /// <summary>
+        /// Occurs when form is resized.
+        /// </summary>
+        /// <param name="sender">Reference to resized control.</param>
+        /// <param name="e">Provides data for the resize event.</param>
+        private void FormResized(object sender, EventArgs e)
+        {
+            if (this.foundFilteredElements)
             {
-                this.dataSourceStream.Dispose();
-                this.dataSourceStream = null;
+                this.textBox.Width = this.Width;
             }
         }
+
+        /// <summary>
+        /// Adds a Web Browser Control to Control Collection.
+        /// </summary>
+        /// <param name="svgData">Svg to display on Browser Control.</param>
+        private void AddBrowserControl(string svgData)
+        {
+            this.browser = new WebBrowser();
+            this.browser.DocumentText = svgData;
+            this.browser.Dock = DockStyle.Fill;
+            this.browser.IsWebBrowserContextMenuEnabled = false;
+            this.browser.ScriptErrorsSuppressed = true;
+            this.browser.ScrollBarsEnabled = true;
+            this.Controls.Add(this.browser);
+        }
+
+        /// <summary>
+        /// Adds a Text Box in Controls for showing information about blocked elements.
+        /// </summary>
+        private void AddTextBoxControl()
+        {
+            this.textBox = new RichTextBox();
+            this.textBox.Text = "Some elements have been blocked to help prevent the sender from identifying your computer. Open this item to view all elements.";

```

Comment:
We should store this in a resource file as this string would have to be localized in the future.